.github/workflows/sdk-python.yml

```diff
@@ -11,7 +11,7 @@ on:
     branches:
       - main
     paths:
-      - 'sdks/python/**'
+      - 'sdks/python/pyproject.toml'
```

Comment:
Important: I think this is a reasonable way to only run in the correct cases on main, since you need to change this file (to bump the version) to publish anyways, and if we're not going to publish then no reason to leave it unchanged.

it's not perfect b/c you could e.g. change a dependency and not the package version, but it should be close at least